tests/client/test_query_params.py

```diff
@@ -0,0 +1,13 @@
+from httpx import Client, QueryParams
+
+
+def test_client_query_params():
+    client = Client(params={"a": "b"})
+    assert isinstance(client.params, QueryParams)
+    assert client.params["A"] == "b"
+
+
+def test_client_query_params_string():
```

Comment:
We'll additionally want some functional tests to make sure the params are being sent over the wire as we expect.

Reply:
Ooh yeah good call, also noticed that queryparams is both `_list` and `_dict`. so I need to do the same indexing as `Headers` does